### PR TITLE
feat(worker): execute worker hooks

### PR DESCRIPTION
**Pull request by fsamin**
Signed-off-by: francois  samin <francois.samin@corp.ovh.com>

### CODE REVIEW

File: engine/worker/internal/run.go [hunk 1]
```diff
@@ -627,6 +685,135 @@ func (w *CurrentWorker) ProcessJob(jobInfo sdk.WorkflowNodeJobRunData) (res sdk.
...
+			hookFilename := fmt.Sprintf("%d-%s-%s", hookConfig.Priority, integrationName, slug.Convert(hookConfig.Label))
+			hookFilePath := path.Join(workingDir, "setup", hookFilename)
+			log.Info(ctx, "setting up hook %q", hookFilePath)
+
+			hookFile, err := fs.Create(hookFilePath)
+			if err != nil {
+				return errors.Errorf("unable to open hook file %q in %q: %v", hookFilePath, w.basedir.Name(), err)
+			}
+			if _, err := hookFile.WriteString(hookConfig.Setup); err != nil {
+				return errors.Errorf("unable to setup hook %q: %v", hookFilePath, err)
```
Comment (sguiheux) [inline engine/worker/internal/run.go:735]: hookFile.Close?

File: engine/worker/internal/run.go [hunk 2]
```diff
@@ -627,6 +685,135 @@ func (w *CurrentWorker) ProcessJob(jobInfo sdk.WorkflowNodeJobRunData) (res sdk.
...
+			}
+
+			hookFilePath = path.Join(workingDir, "teardown", hookFilename)
+			hookFile, err = fs.Create(hookFilePath)
+			if err != nil {
+				return errors.Errorf("unable to open hook file %q: %v", hookFilePath, err)
+			}
+			if _, err := hookFile.WriteString(hookConfig.Teardown); err != nil {
+				return errors.Errorf("unable to setup hook %q: %v", hookFilePath, err)
```
Comment (sguiheux) [inline engine/worker/internal/run.go:747]: hookFile.close?

**Review by sguiheux — CHANGES_REQUESTED** [review]

File: engine/worker/internal/run.go (hunk 1 above)
Comment (fsamin) [reply engine/worker/internal/run.go:735]: yes, just below

Comment (fsamin) [reply engine/worker/internal/run.go:735]: yes just below

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#16765.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Review by yesnault — APPROVED** [review]

**Review by sguiheux — APPROVED** [review]

**Pull request merged**